src/axom/sidre/tests/spio/spio_parallel.cpp

```diff
@@ -953,6 +955,86 @@ TEST(spio_parallel, parallel_decrease_procs)
 
 }
 
+TEST(spio_parallel, sidre_simple_blueprint_example)
+{
+  int my_rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
+  int num_ranks;
+  MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);
+
+  EXPECT_TRUE(num_ranks > my_rank);
+
+#ifdef AXOM_USE_HDF5
+  /*
+   This tests creates a simple sidre + mesh blueprint example.
+
+   It is run with 4 MPI tasks, each task creates a domain
+   with a very small uniform grid. 
+
+  */
+
+  // create a mesh using conduit's examples
+  axom::sidre::Node n_mesh;
+  // create a 2x2 element uniform mesh for each domain
+  conduit::blueprint::mesh::examples::basic("uniform",
+                                            3,3,1,
+                                            n_mesh);
+
+  // shift example mesh in x using rank to create non-overlapping domains
+  n_mesh["coordsets/coords/origin/x"] = my_rank * 20.0;
+
+  // add an element-assoced field that contains the rank
+  n_mesh["fields/rank/association"] = "element";
+  n_mesh["fields/rank/topology"] = "mesh";
+  n_mesh["fields/rank/values"].set(conduit::DataType::int64(4));
+
+  // fill rank field values
+  axom::int64 *rank_vals_ptr = n_mesh["fields/rank/values"].value();
+
+  for(int i=0;i<4;i++)
+  {
+    rank_vals_ptr[i] = my_rank;
+  }
+
+  // setup the data store
+  DataStore ds;
+  Group* root = ds.getRoot();
+  Group *domain_root = root->createGroup("mesh");
+  // setup sidre group using the conduit tree
+  domain_root->importConduitTree(n_mesh);
+
+  // use SPIO to write to a set of hdf5 files
+  // NOTE:
+  // I (cyrush) tried using json protocols but hit issues in VisIt.
+  // VisIt is tested against general json bp files, but not sidre
+  // flavored json, something we should look into. 
+  axom::sidre::IOManager writer(MPI_COMM_WORLD);
+  const std::string file_name = "out_spio_blueprint_example";
+  writer.write(root, 4, file_name, "sidre_hdf5");
+
+  // make sure all tasks are done writing before we try
+  // to add the blueprint index to the root file
+  MPI_Barrier(MPI_COMM_WORLD);
+  // rank 0 adds the bp index on
+  if(my_rank == 0)
+  {
+    // NOTE:
+    // I (cyrush) used conduit here to keep things simple.
+    // To use sidre, I would first have a conduit bp index tree (like here)
+    // then create a new data store, import the conduit tree into the
+    // new sidre data store, and finally use writeGroupToRootFile().
```

Comment:
Shouldn't we be using ``DataStore::generateBlueprintIndex()``  for this?

https://github.com/LLNL/axom/blob/81f88f60729b0bdf9923a02df8e0ac5e9f6f7e49/src/axom/sidre/core/DataStore.hpp#L376-L402

Reply:
I think we would actually want `writeBlueprintIndexToRootFile`, I missed this since there is is no unit test that exercises this only this example:

https://github.com/LLNL/axom/blob/0f1862c67f152281ba28c1968c445bc597f52c2a/src/axom/sidre/examples/sidre_createdatastore.cpp#L590

that would simplify things and serve as a better example

Reply:
Thanks @cyrush -- yes, that's the one I was looking for!

Reply:
i switched to use the spio helper -- I'll merge pending CI saying things are ok